sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -317,106 +364,96 @@ public int getSize(Message amqpMessage) {
+        byte[] bytes = null;
         final Section body = amqpMessage.getBody();
         if (body != null) {
             //TODO (conniey): Support other AMQP types like AmqpValue and AmqpSequence.
             if (body instanceof Data) {
                 final Binary messageData = ((Data) body).getValue();
-                final byte[] bytes = messageData.getArray();
-                brokeredMessage = new ServiceBusReceivedMessage(bytes);
+                bytes = messageData.getArray();
             } else {
                 logger.warning(String.format(Messages.MESSAGE_NOT_OF_TYPE, body.getType()));
-                brokeredMessage = new ServiceBusReceivedMessage(EMPTY_BYTE_ARRAY);
+                bytes = EMPTY_BYTE_ARRAY;
             }
         } else {
             logger.warning(String.format(Messages.MESSAGE_NOT_OF_TYPE, "null"));
-            brokeredMessage = new ServiceBusReceivedMessage(EMPTY_BYTE_ARRAY);
+            bytes = EMPTY_BYTE_ARRAY;
         }
+        final ServiceBusReceivedMessage brokeredMessage = new ServiceBusReceivedMessage(bytes);
+        AmqpAnnotatedMessage brokeredAmqpAnnotatedMessage = brokeredMessage.getAmqpAnnotatedMessage();
 
         // Application properties
         ApplicationProperties applicationProperties = amqpMessage.getApplicationProperties();
         if (applicationProperties != null) {
             final Map<String, Object> propertiesValue = applicationProperties.getValue();
-            brokeredMessage.getProperties().putAll(propertiesValue);
-
-            if (propertiesValue.containsKey(DEAD_LETTER_REASON)) {
-                brokeredMessage.setDeadLetterReason(String.valueOf(propertiesValue.get(DEAD_LETTER_REASON)));
-            }
-            if (propertiesValue.containsKey(DEAD_LETTER_DESCRIPTION)) {
-                brokeredMessage.setDeadLetterErrorDescription(String.valueOf(
-                    propertiesValue.get(DEAD_LETTER_DESCRIPTION)));
-            }
+            brokeredAmqpAnnotatedMessage.getApplicationProperties().putAll(propertiesValue);
         }
 
         // Header
-        brokeredMessage.setTimeToLive(Duration.ofMillis(amqpMessage.getTtl()));
-        brokeredMessage.setDeliveryCount(amqpMessage.getDeliveryCount());
+        final AmqpMessageHeader brokeredHeader = brokeredAmqpAnnotatedMessage.getHeader();
+        brokeredHeader.setTimeToLive(Duration.ofMillis(amqpMessage.getTtl()));
+        brokeredHeader.setDeliveryCount(amqpMessage.getDeliveryCount());
+        brokeredHeader.setDurable(amqpMessage.getHeader().getDurable());
+        brokeredHeader.setFirstAcquirer(amqpMessage.getHeader().getFirstAcquirer());
+        brokeredHeader.setPriority(amqpMessage.getPriority());
+
+        // Footer
+        final Footer footer = amqpMessage.getFooter();
+        if (footer != null && footer.getValue() != null) {
+            @SuppressWarnings("unchecked") final Map<Symbol, Object> footerValue = footer.getValue();
```

Comment:
Is there a reason we can't use a foreach loop for this?

Reply:
I am using `setValues(Map<Symbol, Object> sourceMap, Map<String, Object> targetMap)` which is foreach loop.